File: kmp/shared/src/commonMain/kotlin/net/yumnumm/eqmonitor/core/data/network/model/enum/OriginTimePrecision.kt
package net.yumnumm.eqmonitor.core.data.network.model.enum

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class OriginTimePrecision {
    @SerialName("month")
    MONTH,
    @SerialName("day")
    DAY,
    @SerialName("hour")
    HOUR,
    @SerialName("minute")
    MINUTE,
    @SerialName("second")
    SECOND,
    @SerialName("millisecond")
    MILLISECOND;
}

File: app/lib/feature/earthquake_history/data/model/origin_time_precision.dart
import 'package:eqmonitor_api/eqmonitor_api.dart' as api;
import 'package:freezed_annotation/freezed_annotation.dart';

@JsonEnum(fieldRename: FieldRename.screamingSnake)
enum OriginTimePrecision {
  millisecond,
  second,
  minute,
  hour,
  day,
  month,
}

extension OriginTimePrecisionApiExtension on api.OriginTimePrecision {
  OriginTimePrecision get toOriginTimePrecision => switch (this) {
    .millisecond => .millisecond,
    .second => .second,
    .minute => .minute,
    .hour => .hour,
    .day => .day,
    .month => .month,
  };
}

extension OriginTimePrecisionToApiExtension on OriginTimePrecision {
  api.OriginTimePrecision get toApiOriginTimePrecision => switch (this) {
    .millisecond => .millisecond,
    .second => .second,
    .minute => .minute,
    .hour => .hour,
    .day => .day,
    .month => .month,
  };
}
